calculator.js

```diff
@@ -4,6 +4,45 @@ let lastButtonClicked = null;
 
 let button = document.getElementById("result");
 
+let numBtnElem = document.querySelectorAll("[data-number]")
+    // numBtnElem.forEach(function(button) {
+    // button.addEventListener('click', function() {
+    //         (lastButtonClicked === "Operator") result.textContent = "";
+    //         const currentValueOfResult = result.textContent;
+    //         const newValue = currentValueOfResult + numBtnElem.dataset.number;
```

Comment:
``` const newValue = currentValueOfResult + numBtnElem.dataset.number;```
This should have worked for you, but you're using the wrong object

`numBtnElem` is a collection of HTML elements.
You're using `forEach` to loop over this collection and you're passing a function to the `forEach` function
this function is passed a single element which you're adding event listener to it on line 9
you should have used this element to access the dataset property